doc/getting-started/tutorial.md

```diff
@@ -0,0 +1,149 @@
+```kotlin
+apply {
+    plugin("kotlin")
+    plugin("application")
+}
+```
+
+
+## Configuring A Gradle Plugin
+
+Once a Gradle plugin is *applied* it is configured after the **apply** block. A call is made to the **configure** function with the plugin passed through as the generic type, eg:
+```kotlin
+configure<ApplicationPluginConvention> {
+    mainClassName = "org.example.codekata.MainKt"
+}
+```
+
+Note that some plugins need to be imported first before they can be configured.
+
+
+## Adding Dependencies
+
+Plugin dependencies are added in the **dependencies** block directly inside the **buildscript** block, eg:
+```kotlin
+//...
+buildscript {
+  //...
+    repositories {
+        jcenter()
+    }
+
+    dependencies {
+        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${extra["kotlin-ver"]}")
+        classpath("org.jetbrains.dokka:dokka-gradle-plugin:${extra["dokka-ver"]}")
+    }
+}
+```
+
+Project dependencies are added in the **dependencies** block, eg:
+```kotlin
+dependencies {
+    compile("org.jetbrains.kotlin:kotlin-stdlib-jre8:${extra["kotlin-ver"]}")
+    testCompile("junit:junit:${extra["junit-ver"]}")
+}
+```
+
+## Gradle Tasks
+
+Gradle can run tasks which can be thought of as build actions (eg **compileKotlin** - Compile Kotlin source files). Usually when you run Gradle a task is specified that is to be executed when running a build. It is good practice when customising/creating multiple Gradle tasks to do it within the **tasks** block, which is positioned after the **dependencies** block.
+
+
+### Customising A Task
+
+Often you will find it useful to customise some Gradle tasks (ones supplied by Gradle and/or a third party Gradle plugin). In order to customise a task it needs to be fetched first (via a lookup), eg:
+
+```kotlin
+tasks {
+    // Fetching the "compileKotlin" task and customising it.
+    "compileKotlin"(KotlinCompile::class) {
+        doLast { println("Finished compiling.") }
+    }
+}
+```
+
+
+Alternatively you can just fetch the task if it is going to be used in multiple places in a build script, eg:
+
+```kotlin
+tasks {
+    val compileKotlin = get("compileKotlin") as KotlinCompile
+    
+    compileKotlin.doFirst { println("Warming up the Kotlin compiler...") }
+    compileKotlin.doLast { println("Finished compiling.") }
+}
+```
+
+
+### Creating A Custom Task
+
+When creating a custom Gradle task it needs to be given a unique name, and if its based on a task type (eg Jar) the class object representing the type, eg:
+
+```kotlin
+// ...
+
+tasks {
+    // Create a custom task called "createDokkaJar" that is based on the Jar type, and depends on 
+    // the "dokka" task to run first before executing.
+    "createDokkaJar"(Jar::class)" {
+        dependsOn("dokka")
+        classifier = "javadoc"
+        from(dokka.outputDirectory)
+    }
+}
+```
```

Comment:
Can you add something mentioning:
```kotlin
val createDokkaJar = task<Jar>("createDokkaJar") {
    dependsOn("dokka")
    classifier = "javadoc"
    from(dokka.outputDirectory)
}
```
as also being a valid way of creating the task?